system/HTTP/IncomingRequest.php

```diff
@@ -152,14 +152,19 @@ class IncomingRequest extends Request
 	 * @param string                      $body
 	 * @param \CodeIgniter\HTTP\UserAgent $userAgent
 	 */
-	public function __construct($config, $uri = null, $body = 'php://input', UserAgent $userAgent)
+	public function __construct($config, $uri = null, $body = 'php://input', UserAgent $userAgent = null)
```

Comment:
prefer to require UserAgent in this request. It it already added in the service method, so revert this change.

Reply:
Changed it because there is a concept, always keep optional parameter the end of parameter listing in function arguments. Hence changed.

Reply:
I'm familiar with the concept. In this case, it doesn't work, though. Due to the different types of rules, and some wanting/needing more information than others, the first item should always be the $str. And sometimes that value will be null.